src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/Example1.java

```diff
@@ -0,0 +1,27 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalIdentifierName"/>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.illegalidentifiername;
+
+// xdoc section -- start
+public class Example1 {
+  public static void main(String... args) {
+    var var = 4; // violation
+    int record = 15; // violation
+    String yield = "yield"; // violation
+
+    record Record (Record record){} // 2 violations
+
+    String yieldString = "yieldString"; // ok, part of another word
+    record MyRecord(){} // ok, part of another word
+    var variable = 2; // ok, part of another word
+    String _; // violation
```

Comment:
I think the error being thrown is due to the usage of '` _ `' as an identifier. 
This has been deprecated since release 9, I believe.

Reply:
![image](https://github.com/checkstyle/checkstyle/assets/99334169/dc095c6a-78a5-44cf-8887-b9e0755b9060)

"Yes, I confirm that the error is due to '_' being used as an identifier. I initially thought placing the sample file in a non-compilable location would solve the problem. However, Example1.txt suggests placing it in Example1.java. How can I fix this? Do you suggest following the example image and adding a comment like: "// ok, UNDERSCORE is not in configured tokens?"